radar-auth/src/main/java/org/radarcns/auth/authentication/TokenValidator.java

```diff
@@ -150,10 +150,9 @@ public RadarToken validateAccessToken(String token) throws TokenValidationExcept
                 }
                 return new JwtRadarToken(jwt);
             } catch (SignatureVerificationException sve) {
-                LOGGER.warn("Client presented a token with an incorrect signature, fetching public "
-                        + "keys again. Token: {}", token);
+                LOGGER.debug("Client presented a token with an incorrect signature. Token: {}",
+                        token);
                 refresh();
-                return validateAccessToken(token);
```

Comment:
This should set a boolean: if keys were refreshed but no validator validated the token, maybe they will succeed with the new keys. This retry cycle can happen only once.

Reply:
Alternatively, add a private method with an additional “tryRefresh” boolean. This will be true in the first call, and false in the recursive call.